libs/platform/router/lit/lit-router.ts

```diff
@@ -97,12 +125,60 @@ export class LitRouter extends Router {
       return;
     }
 
-    this.urlSearchParams = Object.fromEntries(
-      new URLSearchParams(
-        decodeURIComponent(globalThis.location?.search)
-      ).entries()
-    );
+    this.storeUrlSearchParams();
 
     await super.goto(pathname);
   }
+
+  private _onClick = (e: MouseEvent) => {
+    const isNonNavigationClick =
+      e.button !== 0 || e.metaKey || e.ctrlKey || e.shiftKey;
+    if (e.defaultPrevented || isNonNavigationClick) {
+      return;
+    }
+
+    const anchor = e
+      .composedPath()
+      .find((n) => (n as HTMLElement).tagName === 'A') as
+      | HTMLAnchorElement
+      | undefined;
+    if (
+      anchor === undefined ||
+      anchor.target !== '' ||
+      anchor.hasAttribute('download') ||
+      anchor.getAttribute('rel') === 'external'
+    ) {
+      return;
+    }
+
+    const href = anchor.href;
+    if (href === '' || href.startsWith('mailto:')) {
+      return;
+    }
+
+    const location = window.location;
+    if (anchor.origin !== origin) {
+      return;
+    }
+
+    e.preventDefault();
+    if (href !== location.href) {
+      window.history.pushState(
+        { counter: this.routerService.getCounter() + 1 },
```

Comment:
Can't we put conter here? We can leave it for the merging routing followup and unit test coverage.